src/RamRelation.h

```diff
@@ -37,63 +30,33 @@ namespace souffle {
  * @brief A RAM Relation in the RAM intermediate representation.
  */
 class RamRelation : public RamNode {
-protected:
-    /** Name of relation */
-    const std::string name;
-
-    /** Arity, i.e., number of attributes */
-    const size_t arity;
-
-    /** Number of height parameters for provenance */
-    const size_t numberOfHeights;
-
-    /** Name of attributes */
-    const std::vector<std::string> attributeNames;
-
-    /** Type of attributes */
-    const std::vector<std::string> attributeTypeQualifiers;
-
-    /** Data-structure representation */
-    const RelationRepresentation representation;
-
 public:
-    /** Used by Interpreter only */
-    mutable void* relation = nullptr;
-
     RamRelation(const std::string name, const size_t arity, const size_t numberOfHeights,
-            const std::vector<std::string> attributeNames,
-            const std::vector<std::string> attributeTypeQualifiers,
-            const RelationRepresentation representation)
-            : name(std::move(name)), arity(arity), numberOfHeights(numberOfHeights),
-              attributeNames(std::move(attributeNames)),
-              attributeTypeQualifiers(std::move(attributeTypeQualifiers)), representation(representation) {
-        assert(this->attributeNames.size() == arity || this->attributeNames.empty());
-        assert(this->attributeTypeQualifiers.size() == arity || this->attributeTypeQualifiers.empty());
+            const std::vector<std::string> attribs, const std::vector<std::string> typs,
```

Comment:
`attribs` -> `attributeNames`
`typs` -> `attributeTypes`

Reply:
Again, we don't want same names for arguments/member variables.